shared/src/commonMain/kotlin/org/hyperskill/app/main/presentation/AppFeature.kt

```diff
@@ -25,8 +31,15 @@ interface AppFeature {
             val isAuthorized: Boolean,
             val isMobileLeaderboardsEnabled: Boolean,
             internal val streakRecoveryState: StreakRecoveryFeature.State = StreakRecoveryFeature.State(),
-            internal val welcomeOnboardingState: WelcomeOnboardingFeature.State = WelcomeOnboardingFeature.State()
-        ) : State
+            internal val welcomeOnboardingState: WelcomeOnboardingFeature.State = WelcomeOnboardingFeature.State(),
+            internal val subscriptionType: SubscriptionType? = null,
+            internal val appShowsCount: Int = 1
+        ) : State {
+            internal val shouldShowPaywall: Boolean
```

Comment:
I don't see the need for a `shouldShowPaywall` extension - we are using it only in the `AppReducer` let's move `APP_SHOWS_COUNT_TILL_PAYWALL` and `shouldShowPaywall` to reducer.

_For me feature is just the interface of the feature data flow, it should not contain business logic._